expr/literals.go

```diff
@@ -220,6 +248,17 @@ func (*NestedLiteral[T]) isRootRef() {}
 func (t *NestedLiteral[T]) String() string {
 	return fmt.Sprintf("%s(%v)", t.Type, t.Value)
 }
+func (t *NestedLiteral[T]) ValueString() string {
+	switch x := any(t.Value).(type) {
+	case ListLiteralValue:
+		var items []string
+		for _, item := range x {
+			items = append(items, item.ValueString())
+		}
+		return fmt.Sprintf("[%s]", strings.Join(items, ", "))
```

Comment:
why not use `strings.Builder` and build the result directly instead of using the intermediate slice?

Reply:
I used to use that style (building up the string and adding the separator as I visited the elements).  I switched to this style to avoid the is the first element check needed to prevent adding  the duplicate separator.  It may not perform as well but the two activities -- gathering the data and managing the presentation -- are separate.

That said I have no strong preference.  If you'd like it the other way I can make that change.

Reply:
Honestly I mostly agree with you, the only time it would matter is if the number of items is particularly large and they are doing this frequently. 

I say leave it like this for now untill someone comes and complains about memory or performance and shows a use case that really needs it